### ISSUE TITLE
Missing destination host

### ISSUE THREAD

**Issue by vbohata** [elastic/ecs#500]
We have a licensing server with its logs containing license id, user name and host to which the license has been assigned. The host is not always domain or ip, it can be some other name. So host.name is the best field for this. But as host refers to the licensing server itself, the better field for this would be destination.host.name.
But host.* is not expected to be nested under the destination unlike user.*.  What field should be used instead of destination.host.name? Should not be source/destination/host modified so host could be nested the same/consistent way as the user can?

**Comment by webmat** [comment]
At this time, source/destination and client/server are meant to represent the participants in a network exchange.

I understand how semantically, using `destination.*` to describe the intended destination of a license would make sense. But I think it may be confusing to use the field set in this way, especially if this doesn't apply to the other 3 field sets.

For now I would suggest tracking that in a custom field. I assume you're already tracking the license ID in a custom field? Perhaps a sibling field to it would make sense to record the destination as well?

I'm curious to see the full schema for one of these events, if you'd like to share it. It would be very useful to consider, when we work on modelling various aspects of asset inventory in ECS.

**Comment by vbohata** [comment]
Yes, for license ID I use custom field. OK for now I will use a custom field.
The logs are being generated by license4j floating license server. License related log lines look like this:

`25 Mar 2019 11:49:50,827 - License assigned. (license id: 1087805    product id: MY_PRODUCT    edition: null    version: null    user: my.user    host: my.host.name) (2 of 100 license in use)`

**Comment by vbohata** [comment]
Btw. this issue is probably also related to old/new for user name etc. Now it is impossible to track old/new user names ... using the ECS. Maybe following top fields could help to both as they are very general:
- origin (user)
- target (user, host ...)

**Comment by webmat** [comment]
Yes, your issue also made me think of modelling "affected" or "target" user. For licenses, `target.host` would work well. 🤔 

It's on the short list of things we want to work on. Now that [some other big projects](https://www.elastic.co/blog/migrating-to-elastic-common-schema-in-beats-environments) are behind us, work on ECS will pick up again a little more ;-)
